### PR TITLE
Toggle terms for save payments in UPE block checkout

### LINKED ISSUE #2605: Add terms parameter to UPE when toggled by save payment checkbox in Blocks
This was long thought to be the final piece of the puzzle of supporting saved payment methods. However, I've now discovered that if this terms flag is set to `always` for a payment method, then this text is displayed beneath the payment method inside the UPE whenever it is selected.

```
By providing your card information, you allow YOUR.STORE to charge your card for future payments in accordance with their terms.
```

This text should be displayed if a customer decides to select the checkbox to save their payment method for future purchases. This parameter should be set at all times at checkout, if the cart contains a subscription item. Otherwise this parameter should be toggled for each supported payment method at checkout, as the save payment method checkbox is toggled. 

Within this #2531, this property has complete support during the regular checkout; however, during a Blocks checkout, this value will only be set if there is a subscription product in a customer's cart. This is because I couldn't figure out where the saved payment checkbox was rendered in React and consequently don't know how to put an `onChange` event on that component.

### PR DESCRIPTION
**Pull request by malithsen**
Fixes #2605

#### Changes proposed in this Pull Request

This change adds the legal mandate text to the UPE block checkout when the "Save payment information to my account for future purchases." checkbox is selected. The mandate should always be shown when a subscription product is in the cart. Otherwise, it should be shown when the save payment method checkbox is toggled.

Screen Shot 2021-11-11 at 4 08 55 PM

#### Testing instructions

1. Enable the UPE at checkout using the WCPay Dev Tools.
2. Add the WooCommerce Blocks plugin
3. As a customer, add a subscription product to the cart and proceed to a blocks checkout. Note that the above legal mandate is present underneath the payment fields inside the UPE.
4. Remove any subscription products from the cart and add a regular product. Proceed to the blocks checkout and note that when the save payment checkbox is checked, the legal mandate appears. Confirm that this mandate disappears if the checkbox is unchecked

-------------------

- [x] Added changelog entry (or does not apply)
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Added testing instructions to the [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) (or does not apply)

### CODE REVIEW

**Review by chickenn00dle — COMMENTED** [review]
Nice work @malithsen 🙌 

I tested and this fixes the issue of missing terms when save payment method checkbox is selected.

I just had one question before approving this. I noticed that when a subscription product is being checked out, the save payment method checkbox becomes redundant. Like with regular checkout, it would make sense to hide this when checking out a subscription product as it's implied the payment method will be saved: https://cloudup.com/cZnVJVuDboG

That said, I haven't really looked too far into working with the checkout block, so am not too familiar with what is possible here. Is this something that we can do here?

This might also be something we can open a separate issue for and work on down the line.

Let me know what you think.

**Comment by malithsen** [comment]
Good point @chickenn00dle I updated it in https://github.com/Automattic/woocommerce-payments/pull/3349/commits/6f74e19002c7bd10a4e18547f1874d5b71cb67b6. Thank you for the review

File: client/checkout/blocks/upe.js [hunk 1]
```diff
@@ -45,7 +45,10 @@ registerPaymentMethod( {
 	ariaLabel: __( 'WooCommerce Payments', 'woocommerce-payments' ),
 	supports: {
 		showSavedCards: getConfig( 'isSavedCardsEnabled' ) ?? false,
-		showSaveOption: getConfig( 'isSavedCardsEnabled' ) ?? false,
+		showSaveOption:
+			( getConfig( 'isSavedCardsEnabled' ) &&
+				! getConfig( 'cartContainsSubscription' ) ) ??
+			false,
```
Comment (chickenn00dle) [inline client/checkout/blocks/upe.js:51]: Is the `?? false` bit here redundant? The `&&` operator should result in either true or false if either condition is falsey, which should be enough I think.

**Review by chickenn00dle — CHANGES_REQUESTED** [review]
Awesome! Thanks for taking care of that @malithsen 👍 

I left just one small nitpick and then this is good to go.

File: client/checkout/blocks/upe.js (hunk 1 above)
Comment (malithsen) [reply client/checkout/blocks/upe.js:51]: > Is the ?? false bit here redundant?

Hi @chickenn00dle `getConfig` could technically return a `null` https://github.com/Automattic/woocommerce-payments/blob/develop/client/utils/checkout.js#L9, `null && null` would still be a `null`. I am not sure of the chances of these configs actually not being set though.

Comment (chickenn00dle) [reply client/checkout/blocks/upe.js:51]: Thanks for clarifying. This makes sense 👍

**Review by chickenn00dle — APPROVED** [review]

**Pull request merged**